skafka/src/main/scala/com/evolutiongaming/skafka/Blocking.scala

```diff
@@ -1,14 +1,30 @@
 package com.evolutiongaming.skafka
 
+import cats.effect.{Async, ContextShift}
+import cats.implicits._
+import com.evolutiongaming.catshelper.FromFuture
+
 import scala.concurrent.{ExecutionContext, Future}
 
-trait Blocking[F[_]] {
-  def apply[T](f: => T): F[T]
-}
+final case class Blocking(ec: ExecutionContext)
 
 object Blocking {
+  private[skafka] def blocking[F[_] : ContextShift, A](f: F[A])(implicit b: Blocking): F[A] =
+    ContextShift[F].evalOn(b.ec)(f)
+
+  private[skafka] def blocking[F[_] : Async : ContextShift, A](thunk: => A)(implicit b: Blocking): F[A] =
+    blocking {
+      Async[F].delay(thunk)
+    }
+
+  private[skafka] def blockingS[F[_] : Async : ContextShift, A](thunk: => A)(implicit b: Blocking): F[A] =
+    blocking(thunk)
 
-  def future(ec: ExecutionContext): Blocking[Future] = new Blocking[Future] {
-    def apply[T](f: => T): Future[T] = Future(f)(ec)
-  }
+  private[skafka] def fromFutureBlocking[F[_] : Async : ContextShift, A](f: => Future[A])(implicit b: Blocking): F[A] =
```

Comment:
do we really need this method ?

Reply:
I wanted to make sure the instantiation of future is also done shifted to blocking EC, which doesn't seem to be provided by `FromFuture`.

Reply:
I see now I don't need `Async[F].delay` though.